scipy/signal/filter_design.py

```diff
@@ -1033,6 +1033,24 @@ def zpk2sos(z, p, k, pairing='nearest'):
     return sos
 
 
+def _align_nums(nums):
+    try:
+        nums = asarray(nums)
+        assert np.issubdtype(nums.dtype, np.number)
```

Comment:
`assert` statements are skipped when running `python -O` so they shouldn't be relied upon